### DISCUSSION TITLE
Im trying to install AIO on docker on windows but it fail with this msg:Cannot connect to the docker socket. Cannot proceed.

### DISCUSSION THREAD

**Discussion by TheMrjuanmita** [nextcloud/all-in-one#2436] [Questions] [answered]
i tried everything but this image keep failling and i dont know what can i do more

**Comment by TheMrjuanmita** [comment]
this is the command i write:
docker run ^
--sig-proxy=false ^
--name nextcloud-aio-mastercontainer ^
--restart always ^
--publish 80:80 ^
--publish 8080:8080 ^
--publish 8443:8443 ^
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config ^
--volume //var/run/docker.sock:/var/run/docker.sock:ro ^
-e NEXTCLOUD_DATADIR="/run/desktop/mnt/host/d/Cloud" ^
-e WATCHTOWER_DOCKER_SOCKET_PATH=/var/run/docker.sock ^
nextcloud/all-in-one:latest

**Reply by Zoey2936** [reply]
do you use the wsl2 backend of docker?

**Comment by TheMrjuanmita** [comment]
yep, I installed wsl2 on my windows before docker

**Comment by szaimen** [comment]
I would personally recommend to follow this guide: https://nextcloud.com/blog/your-guide-to-the-nextcloud-all-in-one-on-windows-10-11/

**Comment by TheMrjuanmita** [comment]
i followed the guide but it says the same error

**Comment by szaimen** [comment]
Hm... Did you make sure that Docker Desktop is running before entering the command into the command prompt?

**Comment by TheMrjuanmita** [comment]
yes, in format the pc and followed the guide and say the same error again

**Comment by szaimen** [comment]
Did you enable the wsl2 backend for Docker Desktop and also updated the Linux kernel like shoen in the guide?

**Comment by TheMrjuanmita** [comment]
yes, i did everything like in the guide

**Comment by szaimen** [comment]
I fear I am out of ideas then. It worked in my testing.

You could try to use the All-in-One VM image alternatively, which is linked on the website: https://nextcloud.com/install/#instructions-server

**Comment by TheMrjuanmita** [comment]
thanks for help, i will try the VM option XD

**Comment by szaimen** [comment]
Just to report back, I have the assuption that the Docker Desktop update v4.19 broke things. See https://github.com/nextcloud/all-in-one/issues/2450

**Comment by TheMrjuanmita** [comment]
i downloaded docker 3.6 and i installed nextcloud with a duckdns domain but when is on the container tab, and say all are running, the nexrtcloud log say this Self test failed: Error while communicating with nextcloud instance

**Reply by szaimen** [reply]
Does Nextcloud office work?

**Reply by TheMrjuanmita** [reply]
im starting again containers

**Reply by TheMrjuanmita** [reply]
![image](https://user-images.githubusercontent.com/63239619/235485126-259fd8fe-cd45-470b-86fd-bc882fe9b21f.png)

**Reply by TheMrjuanmita** [reply]
that's the end of the log of collabora

**Reply by TheMrjuanmita** [reply]
i have opened 443 80 and 8443 ports

**Reply by szaimen** [reply]
Is Nextcloud Office opening correclty when you click on a document??

**Reply by TheMrjuanmita** [reply]
i cant open nextcloud when it finish from run containers

**Comment by szaimen** [comment]
Can you follow https://github.com/nextcloud/all-in-one/discussions/2105?

**Reply by TheMrjuanmita** [reply]
i havent that error

**Comment by TheMrjuanmita** [comment]
now i hav installed nextcloud but when i hit the button it didnt show me the web
![image](https://user-images.githubusercontent.com/63239619/235493290-99c22a2f-8dca-4d58-94fb-1375e54f94c7.png)

**Reply by TheMrjuanmita** [reply]
![image](https://user-images.githubusercontent.com/63239619/235493334-33a323b9-b1e5-4e63-95ac-e9dbc726ce98.png)

**Reply by TheMrjuanmita** [reply]
this is from apache
![image](https://user-images.githubusercontent.com/63239619/235493608-6756b14e-4620-4ec0-b09f-d49a8a276c15.png)

**Reply by szaimen** [reply]
Can you connect to your Nextcloud when outisde your LAN?

**Reply by TheMrjuanmita** [reply]
i cant connect, and i have opened ports on my router and on windows firewall

**Reply by szaimen** [reply]
Did you skip the domain validation or did it accept the domain without skipping it?

**Reply by TheMrjuanmita** [reply]
i skiped it bc the port checker said that ports were opened

**Comment by szaimen** [comment]
Did you switch the VM correctly to bridge mode like explained in its readme and forwarded the ports to the ip-address of the VM?

**Reply by TheMrjuanmita** [reply]
no, i'm with docker on windows now, wuth 3.6 version

**Reply by szaimen** [reply]
Ah so you actually were able to make it work with docker desktop even though you werent? I am confused now

**Reply by TheMrjuanmita** [reply]
i werent with docker 4.19 XD

**Reply by TheMrjuanmita** [reply]
but i realised that i cant open 8443 port tcp onj my router

**Comment by szaimen** [comment]
Can you post the output of sudo docker inspect nextcloud-aio-apache here?

**Reply by TheMrjuanmita** [reply]
[
    {
        "Id": "205f82cf151704d8c33a0448c510ed6032cc8c1da721fa1bea2791665cf2b4dc",
        "Created": "2023-05-01T17:00:46.6460747Z",
        "Path": "start.sh",
        "Args": [
            "/usr/bin/supervisord",
            "-c",
            "/supervisord.conf"
        ],
        "State": {
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 23284,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2023-05-01T17:00:50.2549087Z",
            "FinishedAt": "0001-01-01T00:00:00Z",
            "Health": {
                "Status": "unhealthy",
                "FailingStreak": 46,
                "Log": [
                    {
                        "Start": "2023-05-01T17:36:19.8556493Z",
                        "End": "2023-05-01T17:36:41.0523484Z",
                        "ExitCode": 1,
                        "Output": ""
                    },
                    {
                        "Start": "2023-05-01T17:37:11.0845511Z",
                        "End": "2023-05-01T17:37:32.2466939Z",
                        "ExitCode": 1,
                        "Output": ""
                    },
                    {
                        "Start": "2023-05-01T17:38:02.2697336Z",
                        "End": "2023-05-01T17:38:23.4585916Z",
                        "ExitCode": 1,
                        "Output": ""
                    },
                    {
                        "Start": "2023-05-01T17:38:53.4961103Z",
                        "End": "2023-05-01T17:39:14.663397Z",
                        "ExitCode": 1,
                        "Output": ""
                    },
                    {
                        "Start": "2023-05-01T17:39:44.6869627Z",
                        "End": "2023-05-01T17:40:05.9223182Z",
                        "ExitCode": 1,
                        "Output": ""
                    }
                ]
            }
        },
        "Image": "sha256:0ce2b8eb71a58b55d5e7f46d276bd436c767f513617d03b8a95288dd67e33485",
        "ResolvConfPath": "/var/lib/docker/containers/205f82cf151704d8c33a0448c510ed6032cc8c1da721fa1bea2791665cf2b4dc/resolv.conf",
        "HostnamePath": "/var/lib/docker/containers/205f82cf151704d8c33a0448c510ed6032cc8c1da721fa1bea2791665cf2b4dc/hostname",
        "HostsPath": "/var/lib/docker/containers/205f82cf151704d8c33a0448c510ed6032cc8c1da721fa1bea2791665cf2b4dc/hosts",
        "LogPath": "/var/lib/docker/containers/205f82cf151704d8c33a0448c510ed6032cc8c1da721fa1bea2791665cf2b4dc/205f82cf151704d8c33a0448c510ed6032cc8c1da721fa1bea2791665cf2b4dc-json.log",
        "Name": "/nextcloud-aio-apache",
        "RestartCount": 0,
        "Driver": "overlay2",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "",
        "ExecIDs": null,
        "HostConfig": {
            "Binds": [
                "nextcloud_aio_nextcloud:/var/www/html:ro",
                "nextcloud_aio_apache:/mnt/data:rw"
            ],
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "nextcloud-aio",
            "PortBindings": {
                "443/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "443"
                    }
                ]
            },
            "RestartPolicy": {
                "Name": "unless-stopped",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": null,
            "CapDrop": null,
            "CgroupnsMode": "host",
            "Dns": null,
            "DnsOptions": null,
            "DnsSearch": null,
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "private",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": null,
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": null,
            "DeviceCgroupRules": null,
            "DeviceRequests": null,
            "KernelMemory": 0,
            "KernelMemoryTCP": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": false,
            "PidsLimit": null,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware"
            ],
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ]
        },
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/98d4b3f49ee2fb558dfe226e4f3d15c7e47f054cfef1bed5a8a087e8966a30b6-init/diff:/var/lib/docker/overlay2/a020f23779d7a170c2ea928f93642af2ffc8a5007c8b28c92c6744e0878b972e/diff:/var/lib/docker/overlay2/3f6390056a1812b598369ba02543fe62d4400cba266d325477e90cbed8308389/diff:/var/lib/docker/overlay2/250e9c9072680004b083522368d292b1e8041da8f1bab076c63e5a1a4f48bbd1/diff:/var/lib/docker/overlay2/fa62c87bc60afe6f6314456b3b32d304b9c438cb76911cf6e950ca9bf4e53e9e/diff:/var/lib/docker/overlay2/d37c3522c8dc36f2d16e3fb72a6345630e68aa05e311052a7545e35f25565262/diff:/var/lib/docker/overlay2/dea353a9664f78a9dd282c9947801a779ffe0ca532c4d9c4560dbd5b6b9e8d54/diff:/var/lib/docker/overlay2/9a2a92aede5cd59374f47a6118b319cbba67189a2ff02edf52464b10601cd3dd/diff:/var/lib/docker/overlay2/1d931bf78bc55d1206f35d47820f9fd679969535fc53be3939f605142bb3f8b5/diff:/var/lib/docker/overlay2/370605fa25875f6720b35dcecccece256312089ac4bbad31dfd7bc58ff3af650/diff:/var/lib/docker/overlay2/54207e2ebef772cbcc11a0e6a5749c2817f8f94250589f3e4ad1c83fa2aba04b/diff:/var/lib/docker/overlay2/16f68e35c34715a38f0dd0d40326d220d5c055a9ecfff2c26835ada32b221e69/diff:/var/lib/docker/overlay2/fc82d816b7057c5121333e1b5f505a27065b5b991f184b641bd5dd3e93018ab5/diff:/var/lib/docker/overlay2/fe61584a96b394e3de4449fc3452ed0258421ac6e2b5f4d37412cd626f344275/diff:/var/lib/docker/overlay2/25be575eb90a1fb329e297c023ef7b5c292d4253c3b6aacd1382531a7e98b3bd/diff:/var/lib/docker/overlay2/1ea9fc8d15428a07e0344f0c340a3f5cae0cf57e6980608482eb41a4dfa71ac5/diff:/var/lib/docker/overlay2/76585c89fbbcd6c221afe908bb87a4da217180c7c67a58c0569a89436134fba1/diff:/var/lib/docker/overlay2/60b69e72f9bd30b1ceab63b23988849a2e98c2ce77112f90a40facd863a8292c/diff:/var/lib/docker/overlay2/7d01a80817f340e4194e16dfeae2e836d8639d2de4d03785db416376b36b0a2c/diff:/var/lib/docker/overlay2/60df0b389c31da31b69a12c892cae76484b2927ad619e531973e7b61f5dde04e/diff:/var/lib/docker/overlay2/674f00a571349eed8c68ec1ed7e3e32b8a353f40302ba2a1ef11885ef62d24d2/diff:/var/lib/docker/overlay2/2a8970d4f0d77372a98064ae393c1f1ef83617c9f0b2f9a216b91e626eddc32b/diff",
                "MergedDir": "/var/lib/docker/overlay2/98d4b3f49ee2fb558dfe226e4f3d15c7e47f054cfef1bed5a8a087e8966a30b6/merged",
                "UpperDir": "/var/lib/docker/overlay2/98d4b3f49ee2fb558dfe226e4f3d15c7e47f054cfef1bed5a8a087e8966a30b6/diff",
                "WorkDir": "/var/lib/docker/overlay2/98d4b3f49ee2fb558dfe226e4f3d15c7e47f054cfef1bed5a8a087e8966a30b6/work"
            },
            "Name": "overlay2"
        },
        "Mounts": [
            {
                "Type": "volume",
                "Name": "nextcloud_aio_nextcloud",
                "Source": "/var/lib/docker/volumes/nextcloud_aio_nextcloud/_data",
                "Destination": "/var/www/html",
                "Driver": "local",
                "Mode": "ro",
                "RW": false,
                "Propagation": ""
            },
            {
                "Type": "volume",
                "Name": "nextcloud_aio_apache",
                "Source": "/var/lib/docker/volumes/nextcloud_aio_apache/_data",
                "Destination": "/mnt/data",
                "Driver": "local",
                "Mode": "rw",
                "RW": true,
                "Propagation": ""
            }
        ],
        "Config": {
            "Hostname": "205f82cf1517",
            "Domainname": "",
            "User": "www-data",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "443/tcp": {},
                "80/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "NC_DOMAIN=pc-server.duckdns.org",
                "NEXTCLOUD_HOST=nextcloud-aio-nextcloud",
                "COLLABORA_HOST=nextcloud-aio-collabora",
                "TALK_HOST=nextcloud-aio-talk",
                "APACHE_PORT=443",
                "ONLYOFFICE_HOST=nextcloud-aio-onlyoffice",
                "TZ=Europe/Berlin",
                "APACHE_MAX_SIZE=10737418240",
                "APACHE_MAX_TIME=3600",
                "PATH=/usr/local/apache2/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "HTTPD_PREFIX=/usr/local/apache2",
                "HTTPD_VERSION=2.4.57",
                "HTTPD_SHA256=dbccb84aee95e095edfbb81e5eb926ccd24e6ada55dcd83caecb262e5cf94d2a",
                "HTTPD_PATCHES="
            ],
            "Cmd": [
                "/usr/bin/supervisord",
                "-c",
                "/supervisord.conf"
            ],
            "Healthcheck": {
                "Test": [
                    "CMD-SHELL",
                    "healthcheck.sh"
                ]
            },
            "Image": "nextcloud/aio-apache:latest",
            "Volumes": {
                "/mnt/data": {}
            },
            "WorkingDir": "/usr/local/apache2",
            "Entrypoint": [
                "start.sh"
            ],
            "OnBuild": null,
            "Labels": {
                "com.centurylinklabs.watchtower.monitor-only": "true"
            },
            "StopSignal": "SIGWINCH"
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "bcc0549fd664445d722da921d7c62578de56afa6fabb86965d95f2bdc34623fe",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {
                "443/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "443"
                    },
                    {
                        "HostIp": "::",
                        "HostPort": "443"
                    }
                ],
                "80/tcp": null
            },
            "SandboxKey": "/var/run/docker/netns/bcc0549fd664",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "nextcloud-aio": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": [
                        "205f82cf1517"
                    ],
                    "NetworkID": "e4e40cbaded30d1cb2bb67870db86582c3293abf14d9091448a4207cfd64746f",
                    "EndpointID": "bec10219ed181fc1a7d69c1082338cafd31a5d80dd1fb3c78f9c9a1d41ea6c46",
                    "Gateway": "172.18.0.1",
                    "IPAddress": "172.18.0.7",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "02:42:ac:12:00:07",
                    "DriverOpts": null
                }
            }
        }
    }
]

**Comment by szaimen** [comment]
Hm... Are you sure that opening your nextcloud via your public domain from outside your local area network, e.g. by connecting via your mobile internet foes not work?

**Reply by TheMrjuanmita** [reply]
ok, im with a friend on discord and he can acces to nextcloud, now how can i acces locally? XD

**Reply by szaimen** [reply]
> and he can acces to nextcloud,

Nice!

**Comment by szaimen** [comment, marked as answer]
> now how can i acces locally? XD

This is something that you need to fix in your network infrastructure. Most likely your router or firewall is the culprit. 

You can also check if the following helps: https://github.com/nextcloud/all-in-one#how-can-i-access-nextcloud-locally

**Reply by TheMrjuanmita** [reply]
ty for help
